### PR TITLE
Fixed disappearing of selected media in media selection

### LINKED ISSUE #3500: "media_selection" does not display title of medias after adding a new media.
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.5.*, 1.6.*

#### Actual Behavior

After adding a image on a"media_selection"-field the title of already added medias are not displayed
bildschirmfoto 2017-08-28 um 11 42 42

#### Steps to Reproduce

- Add some medias on a "media_selection"-field.
- Save it and reload page. (Everything looks good)
- Add a media on the same "media_selection"-Field. (Now only the title of the new Media is displayed)

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed issues | fixes #3500
| License | MIT

#### What's in this PR?

Give the media overlay the complete media items instead of only the id.

#### Why?

The media selection need all items from the media overlay. If the media selection gives the media overlay only the ids of the current selected items. The media overlay will also only return the id. 

#### Example Usage

When you have medias inside a media selection and add a new one from another collection and never opened the collection where the exist medias are. The thumbnail and title of the exist selected items disappear when adding a new media.

### CODE REVIEW

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js [hunk 1]
```diff
@@ -278,6 +288,7 @@ define([
 
         removeHandler: function(id) {
             var data = this.getData();
+            delete this.items[id];
```
Comment (wachterjohannes) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: would set to undefined `this.items[id] = undefined;` delete has a bad performance (see https://jsperf.com/delete-vs-undefined-vs-null/3).

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 CHANGELOG for Sulu
 ==================
 
+* 1.5.6 (unreleased)
+    * HOTFIX      #-    [MediaBundle]           Fixed disappearing of selected medias
```
Comment (danrot) [inline CHANGELOG.md]: Set correct PR number

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js [hunk 3]
```diff
@@ -242,6 +242,16 @@ define([
             startSelectionOverlay.call(this);
         },
 
+        getItems: function() {
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: Rename to `getItemIds`?

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js [hunk 4]
```diff
@@ -171,9 +170,7 @@ define([
                     el: $container,
                     url: this.options.url,
                     instanceName: this.options.instanceName,
-                    preSelectedIds: _.map(this.getData().ids, function(id) {
-                        return {id: id};
-                    }),
+                    preselected: this.getItems(),
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: From what I see `getItems` returns the same as the previous `map`, why did the name of the property change from `preSelectedIds` to `preselected` then?

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js (hunk 3 above)
Comment (wachterjohannes) [reply src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: but the function returns items ;)

Comment (danrot) [reply src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: But not complete items, so this name is really confusing. Also open for other better suggestions.

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js (hunk 4 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: there is no option `preSelectedIds` in media overlay its just `preselected`: https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/overlay/main.js#L52

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js (hunk 1 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: Setting it to undefined will not delete the key and if you will each over the items the removedIds will still be there. (https://jsfiddle.net/u1mqLu59/1/)

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js (hunk 3 above)
Comment (alexander-schranz) [reply src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: it returns the complete items when the complete items are loaded.

**Comment by wachterjohannes** [comment]
@alexander-schranz there is failing everything :)

**Comment by alexander-schranz** [comment]
@wachterjohannes CI is running

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js [hunk 5]
```diff
@@ -242,6 +242,16 @@ define([
             startSelectionOverlay.call(this);
         },
 
+        getItems: function() {
+            return _.map(this.getData().ids, function(id) {
+                if (this.items[id]) {
+                    return this.items[id];
+                }
+
+                return {id: id};
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/media-selection/main.js]: That's where it doesn't load the entire object, but only the id...

**Pull request merged**